### PR TITLE
Added batching for reads and retries for the most heavy function in backups

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Use MultiRead request and retries for collecting metadata at final stage of backup processing.

### CODE REVIEW

File: src/Backups/IBackupCoordination.h [hunk 1]
```diff
@@ -85,6 +90,20 @@ class IBackupCoordination
 
         /// Position in the archive.
         UInt64 pos_in_archive = static_cast<UInt64>(-1);
+
+        [[ maybe_unused ]] String describe()
```
Comment (alexey-milovidov) [inline src/Backups/IBackupCoordination.h:94]: This format cannot be parsed back, a comment is needed to explain that the method is only suitable for debugging.

File: src/Backups/BackupCoordinationRemote.cpp [hunk 2]
```diff
@@ -486,19 +498,127 @@ void BackupCoordinationRemote::updateFileInfo(const FileInfo & file_info)
...
+            result.push_back(std::move(batch));
+        };
+
+        if (max_batch_size == 0)
+        {
+            move_to_result(vec.begin(), vec.end());
+            return result;
+        }
+
+        for (size_t pos = 0; pos < vec.size(); ++pos)
```
Comment (nikitamikhaylov) [inline src/Backups/BackupCoordinationRemote.cpp:525]: This loop could be rewritten, but I think this already Ok.

File: src/Backups/BackupCoordinationRemote.cpp [hunk 3]
```diff
@@ -486,19 +498,127 @@ void BackupCoordinationRemote::updateFileInfo(const FileInfo & file_info)
...
+        {
+            auto zk = getZooKeeper();
+            non_empty_file_infos_serialized = zk->get(non_empty_file_infos_paths);
+        });
+
+        /// Process non empty files
+        for (size_t i = 0; i < non_empty_file_names.size(); ++i)
+        {
+            FileInfo file_info;
+            file_info = deserializeFileInfo(non_empty_file_infos_serialized.value()[i].data);
```
Comment (alesapin) [inline src/Backups/BackupCoordinationRemote.cpp:614]: Same check here for code here.

**Review by alesapin — APPROVED** [review]

File: src/Backups/BackupCoordinationRemote.cpp [hunk 4]
```diff
@@ -486,19 +498,127 @@ void BackupCoordinationRemote::updateFileInfo(const FileInfo & file_info)
...
+    for (auto & batch : batched_escaped_names)
     {
-        String size_and_checksum = zk->get(zookeeper_path + "/file_names/" + escaped_name);
-        UInt64 size = deserializeSizeAndChecksum(size_and_checksum).first;
-        FileInfo file_info;
-        if (size) /// we don't keep FileInfos for empty files
-            file_info = deserializeFileInfo(zk->get(zookeeper_path + "/file_infos/" + size_and_checksum));
-        file_info.file_name = unescapeForFileName(escaped_name);
-        file_infos.emplace_back(std::move(file_info));
+        std::optional<zkutil::ZooKeeper::MultiGetResponse> sizes_and_checksums;
```
[the remark this answers is not in the document]
Comment (nikitamikhaylov) [reply src/Backups/BackupCoordinationRemote.cpp:555]: MultiGetResponse doesn't have default constructor

[the remark this answers is not in the document]
Comment (antonio2368) [reply src/Backups/BackupCoordinationRemote.cpp:555]: should be easy to extend, I'll do it in a different PR

**Pull request merged**